### DISCUSSION TITLE
How do I use viem in React app?

### DISCUSSION THREAD

**Discussion by AndreinNguyen** [wevm/viem#219] [Question] [answered]
I have some confuse about Viem and Wagmi. Should I use both in a React application or not?

**Comment by jxom** [comment, marked as answer]
[wagmi](https://wagmi.sh) is primarily for **dapps & wallets** – a high-level and stateful interface for Ethereum (React Hooks, Vanilla JS, etc).

viem is primarily for **libraries, tooling, server-side dev, scripting, etc** – a low-level and stateless interface for Ethereum.

I would use wagmi for a React application. Soon, wagmi will port from ethers.js to viem internally.

**Reply by AndreinNguyen** [reply]
Thank you, I got it!!!
